druid/examples/identity.rs

```diff
@@ -44,31 +45,68 @@ const FREEZE_COLOR: Selector<Color> = Selector::new("identity-example.freeze-col
 const UNFREEZE_COLOR: Selector = Selector::new("identity-example.unfreeze-color");
 
 /// Honestly: it's just a color in fancy clothing.
-#[derive(Debug, Clone, Data, Lens)]
+#[derive(Clone, Data)]
 struct OurData {
-    #[data(same_fn = "color_eq")]
     color: Color,
 }
 
-fn color_eq(one: &Color, two: &Color) -> bool {
-    one.as_rgba_u32() == two.as_rgba_u32()
+pub fn main() {
+    let window = WindowDesc::new(make_ui).title(
+        LocalizedString::new("identity-demo-window-title").with_placeholder("Color Freezing Fun"),
+    );
+    let data = OurData {
+        color: Color::BLACK,
+    };
+    AppLauncher::with_window(window)
+        .launch(data)
+        .expect("launch failed");
 }
 
-fn split_rgba(rgba: &Color) -> (u8, u8, u8, u8) {
-    let rgba = rgba.as_rgba_u32();
-    (
-        (rgba >> 24 & 255) as u8,
-        ((rgba >> 16) & 255) as u8,
-        ((rgba >> 8) & 255) as u8,
-        (rgba & 255) as u8,
-    )
+/// A constant `WidgetId`. This may be passed around and can be reused when
+/// rebuilding a widget graph; however it should only ever be associated with
+/// a single widget at a time.
+const ID_ONE: WidgetId = WidgetId::reserved(1);
+
+fn make_ui() -> impl Widget<OurData> {
+    // We can also generate these dynamically whenever we need it.
+    let id_two = WidgetId::next();
+    let id_three = WidgetId::next();
+
+    let mut column = Flex::column().with_flex_child(ColorWell::new(true), 1.0);
+    // This doenst need to be a loop, but it allows us to seperate the creation of the buttons and the colorwell.
+    for &id in &[ID_ONE, id_two, id_three] {
+        // Here we can se the `id` to make sure all the buttons corelate with the colorwell.
+        // We give the colorwell an id, and we use that same id to target our commands to that widget specifically.
+        // This allows us to send commands to only one widget, and not the whole window for example.
+        // In this case, when the buttons are clicked we send a command to the corresponding colorwell.
+        let colorwell = ColorWell::new(false).with_id(id);
+        let pin_button = Button::<OurData>::new("pin").on_click(move |ctx, data, _env| {
```

Comment:
Not that I really mind, but is there a reason you changed freeze/unfreeze to pin/sync?

Reply:
yeah, it was a bit confusing for me to see it as freeze/unfreeze.
By default the colour is grey, "freezing" doesnt really make sense if the colour was static anyways. "pin" pins the colour to whatever the main colorWell is, this is basically the same as freeze which isnt the confusing part.
The "unfreeze" is the thing that really didnt make sense, because then you undo something, you expect it to return to the original state. But it doesnt return to its original static colour, it starts to sync the colour.
I tried to have as little suprises as possible, they distract from the example and may confuse someone when reading it.